packages/tooltip/src/tooltip.css

```diff
@@ -10,59 +10,253 @@
+    left: auto;
+}
+
+:host([placement="bottom-end"]) #tip,
+:host([placement="top-end"]) #tip {
+    inset-inline: auto var(--mod-tooltip-pointer-corner-spacing, var(--spectrum-tooltip-pointer-corner-spacing));
+}
+
+:host([placement="bottom-end"][dir="rtl"]) #tip,
+:host([placement="top-end"][dir="rtl"]) #tip {
+    left: var(--mod-tooltip-pointer-corner-spacing, var(--spectrum-tooltip-pointer-corner-spacing));
+    right: auto;
+}
+
+:host([placement*="left"]) #tip,
+:host([placement*="right"]) #tip {
     inset-block-start: 50%;
     transform: translateY(-50%);
 }
 
-:host([placement*="right"]) #tooltip #tip {
-    clip-path: polygon(50% 50%, 105% 100%, 105% 0);
+#tooltip,
+:host([placement*="top"]) #tooltip {
+    margin-block-end: calc(var(--mod-tooltip-tip-block-size, var(--spectrum-tooltip-tip-block-size)) + var(--mod-tooltip-margin, var(--spectrum-tooltip-margin)));
+}
+
+:host(:not([placement*="top"])) #tooltip {
+    margin-block-end: 0;
+}
+
+:host([placement="top-start"][open]),
+:host([placement="top-end"][open]),
+:host([placement*="top"][open]) #tooltip,
+:host([open]) #tooltip {
+    transform: translateY(calc(var(--mod-tooltip-animation-distance, var(--spectrum-tooltip-animation-distance)) * -1));
+}
+
+:host([placement*="bottom"]) #tooltip {
+    margin-block-start: calc(var(--mod-tooltip-tip-block-size, var(--spectrum-tooltip-tip-block-size)) + var(--mod-tooltip-margin, var(--spectrum-tooltip-margin)));
+}
+
+:host([placement*="bottom"][open]) #tooltip {
+    transform: translateY(var(--mod-tooltip-animation-distance, var(--spectrum-tooltip-animation-distance)));
+}
+
+/** Right placement */
+:host([placement*="right"]) #tooltip {
+    margin-left: calc(var(--mod-tooltip-tip-block-size, var(--spectrum-tooltip-tip-block-size)) + var(--mod-tooltip-margin, var(--spectrum-tooltip-margin)));
+}
+
+:host([placement*="right"][open]) #tooltip {
+    transform: translateX(var(--mod-tooltip-animation-distance, var(--spectrum-tooltip-animation-distance)));
+}
+
+:host([placement*="right"]) #tip {
+    clip-path: polygon(calc(100% - var(--mod-tooltip-tip-height-percentage, var(--spectrum-tooltip-tip-height-percentage))) 50%, calc(100% + var(--mod-tooltip-tip-antialiasing-inset, var(--spectrum-tooltip-tip-antialiasing-inset))) 100%, calc(100% + var(--mod-tooltip-tip-antialiasing-inset, var(--spectrum-tooltip-tip-antialiasing-inset))) 0);
     inset-inline: calc(var(--mod-tooltip-tip-block-size, var(--spectrum-tooltip-tip-block-size)) * -2) 100%;
 }
 
-:host([placement*="left"]) #tooltip #tip {
+:host([placement*="right"][dir="rtl"]) #tip {
+    clip-path: polygon(var(--mod-tooltip-tip-height-percentage, var(--spectrum-tooltip-tip-height-percentage)) 50%, calc(100% + var(--mod-tooltip-tip-antialiasing-inset, var(--spectrum-tooltip-tip-antialiasing-inset))) 100%, calc(100% + var(--mod-tooltip-tip-antialiasing-inset, var(--spectrum-tooltip-tip-antialiasing-inset))) 0);
+    left: auto;
+    right: 100%;
+}
```

Comment:
I think the `clip-path`s here and also for the corresponding `[placement*="left"]` may be the same, it might be possible to eliminate the rtl clip-path. I'm less sure about `inset-inline` vs. `left`/`right` but if we were to do this for real this seems like a good opportunity to start using logical properties.